packages/amplify-gen2-codegen/src/backend/synthesizer.ts

```diff
@@ -115,6 +122,291 @@ export class BackendSynthesizer {
+          false,
+        ),
+      ],
+    );
+  }
+
+  private createUserPoolAssignment(userPoolClient: UserPoolClientType, imports: any[]) {
+    const userPoolAtrributesMap = new Map<string, string>();
+    userPoolAtrributesMap.set('ClientName', 'userPoolClientName');
+    userPoolAtrributesMap.set('ClientSecret', 'generateSecret');
+    userPoolAtrributesMap.set('ReadAttributes', 'readAttributes');
+    userPoolAtrributesMap.set('WriteAttributes', 'writeAttributes');
+    userPoolAtrributesMap.set('RefreshTokenValidity', 'refreshTokenValidity');
+    userPoolAtrributesMap.set('AccessTokenValidity', 'accessTokenValidity');
+    userPoolAtrributesMap.set('IdTokenValidity', 'idTokenValidity');
+    userPoolAtrributesMap.set('RefreshToken', 'refreshToken');
+    userPoolAtrributesMap.set('AccessToken', 'accessToken');
+    userPoolAtrributesMap.set('IdToken', 'idToken');
+    userPoolAtrributesMap.set('AllowedOAuthScopes', 'scopes');
+    userPoolAtrributesMap.set('CallbackURLs', 'callbackUrls');
+    userPoolAtrributesMap.set('LogoutURLs', 'logoutUrls');
+    userPoolAtrributesMap.set('DefaultRedirectURI', 'defaultRedirectUri');
+    userPoolAtrributesMap.set('AllowedOAuthFlowsUserPoolClient', 'disableOAuth');
+    userPoolAtrributesMap.set('EnableTokenRevocation', 'enableTokenRevocation');
+    userPoolAtrributesMap.set('EnablePropagateAdditionalUserContextData', 'enablePropagateAdditionalUserContextData');
+    userPoolAtrributesMap.set('SupportedIdentityProviders', 'supportedIdentityProviders');
+    userPoolAtrributesMap.set('AuthSessionValidity', 'authSessionValidity');
+    userPoolAtrributesMap.set('ExplicitAuthFlows', 'authFlows');
+    userPoolAtrributesMap.set('AllowedOAuthFlows', 'flows');
+
+    const test = factory.createCallExpression(
+      factory.createPropertyAccessExpression(factory.createIdentifier('userPool'), factory.createIdentifier('addClient')),
+      undefined,
+      [factory.createStringLiteral(userPoolClient.UserPoolId!), this.createNestedObjectExpression(userPoolClient, userPoolAtrributesMap)],
+    );
+
+    if (this.importDurationFlag) {
+      imports.push(this.createImportStatement([factory.createIdentifier('Duration')], 'aws-cdk-lib'));
+    }
+
+    if (this.readWriteAttributeFlag || this.oAuthFlag || this.supportedIdentityProviderFlag) {
+      const identifiers = [
+        ...(this.readWriteAttributeFlag ? [factory.createIdentifier('ClientAttributes')] : []),
+        ...(this.oAuthFlag ? [factory.createIdentifier('OAuthScope')] : []),
+        ...(this.supportedIdentityProviderFlag ? [factory.createIdentifier('UserPoolClientIdentityProvider')] : []),
+      ];
+
+      if (identifiers.length > 0) {
+        imports.push(this.createImportStatement(identifiers, 'aws-cdk-lib/aws-cognito'));
+      }
+    }
+
+    return test;
+  }
+
+  private createNestedObjectExpression(object: Record<string, any>, map: Map<string, string>): ts.ObjectLiteralExpression {
```

Comment:
Nice work on making this function generic enough for nested objects. 
Not blocking this PR since Cognito APIs are wildly different from their CDK counterparts. I was thinking for the next iteration on this if we can pass a callback function that can be run inside this function to provide custom behavior. That way it can become a generic function that can be used across categories :)

Reply:
Sure, that's a good idea. I'll update that in the next CR